packages/lit-dev-content/src/code-language-preference.ts

```diff
@@ -30,10 +56,26 @@ export const getCodeLanguagePreference = (): CodeLanguagePreference =>
  * a change event on window.
  */
 export const setCodeLanguagePreference = (
-  preference: CodeLanguagePreference
+  preference: CodeLanguagePreference,
+  force = false
 ): void => {
+  if (!force) {
+    let cancelled = false;
+    const detail: BeforeCodeLanguageChangeDetail = {
+      pendingLanguage: preference,
+      cancel: () => {
+        cancelled = true;
+      },
```

Comment:
Very clever synchronous message passing. It's very nice and avoids any async code.